### ISSUE TITLE
Debug support in VS Code

### ISSUE THREAD

**Issue by seesharper** [scriptcs/scriptcs#1120]
Now that VS Code finally has become extensible, maybe it would be possible to add support for debugging scripts in VS Code?. I use VS Code with an empty project.json file to light up OmniSharp and it works really great with ScriptCS. Being able to debug would be fantastic. Any ideas?

Just for reference here is what I got in VS Code to make Intellisense (OmniSharp) work.

project.json 
```
{
	
}
```

tasks.json

```

// A task runner that calls the scriptcs compiler (scriptcs) and 
// compiles the current opened file.
{
	"version": "0.1.0",
		
	"command": "scriptcs",
	
	// The command is a shell script
	"isShellCommand": true,

	// Show the output window only if unrecognized errors occur. 
	"showOutput": "silent",
	
	// args is the current file to compile
	"args": ["${file}", "-debug"],
	
	// use the standard tsc problem matcher to find compile problems
	// in the output.
	"problemMatcher": "$msCompile"
}
```
Note:  The debug flag is just to enable full debug/stack information when an exception is thrown.

**Comment by adamralph** [comment]
@seesharper thanks for raising this! It would certainly be nice to provide some kind of no-brainer experience for debugging scriptcs in VS Code. I haven't had time to play with it yet myself, but it's now on my todo list :smile:

**Comment by jamesqo** [comment]
+1. This would be great!

**Comment by glennblock** [comment]
This would be great! 

I put some cycles in recently trying to get this to work on Mac, but so far no-dice. On OSX,  currently Code uses Mono for debugging. We rely on Mono's `Evaluator` for x-plat compilation. Based on my testing, it doesn't appear like the the end to end for debugging works with the `Evaluator` i.e. it doesn't appear to pick up breakpoints or #line directives. I chatted with @migueldeicaza on this and he suspected there may be bugs there. This does work today with Roslyn, so once Roslyn is working full xPlat, it may just work. It is possible it works with code on Windows now, assuming code is using the full .NET runtime for debugging when on Windows.

**Comment by eamodio** [comment]
Now that .NET core is RC2 (with x-plat) any chance of resurrecting this?

**Comment by glennblock** [comment]
Yes, but we have a lot of work to do to get to RC2. Once we do there is no
reason this wouldn't work.

**Comment by seesharper** [comment]
Any updates on this?

**Comment by glennblock** [comment]
@seesharper not yet, but we are planing to start work on our CoreCLR support one we have that it should be possible. For now VS Debugging is our only option.

**Comment by seesharper** [comment]
Sounds great. What approach are you taking on .NetCore? Roslyn does not provide the scripting API on .Net Core so what do you see as a solution here. Just curious :)

**Comment by filipw** [comment]
> Roslyn does not provide the scripting API on .Net Core

no, it works on .NET Core

**Comment by seesharper** [comment]
really? , that is really great news 

"Scripting APIs require desktop .NET Framework 4.6+."
https://github.com/dotnet/roslyn/wiki/Scripting-API-Samples

Guess they didn't update the docs?

**Comment by filipw** [comment]
yeah here is a sample https://github.com/filipw/script-config/blob/master/coreclr/ConsoleApp1/src/ScriptConfig/Config.cs

**Comment by seesharper** [comment]
Ah, cool !! Having scriptcs on .NetCore is going to be awesome. Having a cross platform story with debug and intellisense support is going to be a killer. 👍 You guys have a time frame on this?

**Comment by seesharper** [comment]
@filipw 
I threw something together today to see if I could get the debugger working running scripts on .NetCore.

I hit the wall at this exception

"cannot emit debug information for a source text without encoding"

https://github.com/seesharper/scriptrunner

Any thoughts?

**Comment by glennblock** [comment]
Just saw your tweet, do you have it fully working now?

**Comment by seesharper** [comment]
Yeah, I actually got it working. Finally. I had to work around a few limitations in Roslyn, but nothing all that big. I will clean it up and update the repo. I am pretty sure that this will come in handy for scriptcs as well

**Comment by glennblock** [comment]
Nice job! Yes, we'd love your help as we port to Core CLR. We need to start that work!

**Comment by glennblock** [comment]
Have you joined our slack?

**Comment by seesharper** [comment]
Updated the sample application with a minimum to get this working.
https://github.com/seesharper/scriptrunner

**Comment by filipw** [comment]
At the moment debugging works in VS Code with Mono Debugger Extension.
To make it work with C# Extension, it's not that much work - because it now has CLR debugger. 

We just need to make sure we emit portable PDBs. At the moment we only emit portable PDBs on Mono, and full PDBs on Windows. I think the solution would be to just add a flag to the command line entry point that allows us to force a specific PDB format - forcing portable PDBs everywhere would be to aggressive at this point.

**Comment by ivanko2000** [comment]
Has this been abandoned - or is there now a way to debug scriptcs with C# extension?
